packages/backend/src/database/migrations/20250717082932_oauth_server_tokens.ts

```diff
@@ -0,0 +1,96 @@
+        );
+
+        CREATE TABLE IF NOT EXISTS oauth2_authorization_codes (
+            authorization_code VARCHAR(255) PRIMARY KEY,
+            expires_at TIMESTAMP NOT NULL,
+            redirect_uri TEXT NOT NULL,
+            scope TEXT[],
+            client_id VARCHAR(255) NOT NULL REFERENCES oauth2_clients(client_id) ON DELETE CASCADE,
+            user_id INTEGER NOT NULL REFERENCES users(user_id) ON DELETE CASCADE,
+            code_challenge VARCHAR(255),
+            code_challenge_method VARCHAR(10),
+            organization_uuid UUID NOT NULL REFERENCES organizations(organization_uuid) ON DELETE CASCADE,
+            created_at TIMESTAMP NOT NULL DEFAULT NOW()
+        );
+
+        CREATE TABLE IF NOT EXISTS oauth2_access_tokens (
+            access_token VARCHAR(255) PRIMARY KEY,
+            expires_at TIMESTAMP NOT NULL,
+            scope TEXT[],
+            client_id VARCHAR(255) NOT NULL REFERENCES oauth2_clients(client_id) ON DELETE CASCADE,
+            user_id INTEGER NOT NULL REFERENCES users(user_id) ON DELETE CASCADE,
+            organization_uuid UUID NOT NULL REFERENCES organizations(organization_uuid) ON DELETE CASCADE,
+            created_at TIMESTAMP NOT NULL DEFAULT NOW()
+        );
+
+        CREATE TABLE IF NOT EXISTS oauth2_refresh_tokens (
+            refresh_token VARCHAR(255) PRIMARY KEY,
+            expires_at TIMESTAMP NOT NULL,
+            scope TEXT[],
+            client_id VARCHAR(255) NOT NULL REFERENCES oauth2_clients(client_id) ON DELETE CASCADE,
+            user_id INTEGER NOT NULL REFERENCES users(user_id) ON DELETE CASCADE,
+            organization_uuid UUID NOT NULL REFERENCES organizations(organization_uuid) ON DELETE CASCADE,
+            created_at TIMESTAMP NOT NULL DEFAULT NOW()
+        );
+
+        CREATE INDEX IF NOT EXISTS oauth2_authorization_codes_client_id_idx ON oauth2_authorization_codes(client_id);
+        CREATE INDEX IF NOT EXISTS oauth2_authorization_codes_user_id_idx ON oauth2_authorization_codes(user_id);
+        CREATE INDEX IF NOT EXISTS oauth2_authorization_codes_expires_at_idx ON oauth2_authorization_codes(expires_at);
+        
+        CREATE INDEX IF NOT EXISTS oauth2_access_tokens_client_id_idx ON oauth2_access_tokens(client_id);
+        CREATE INDEX IF NOT EXISTS oauth2_access_tokens_user_id_idx ON oauth2_access_tokens(user_id);
+        CREATE INDEX IF NOT EXISTS oauth2_access_tokens_expires_at_idx ON oauth2_access_tokens(expires_at);
+        
+        CREATE INDEX IF NOT EXISTS oauth2_refresh_tokens_client_id_idx ON oauth2_refresh_tokens(client_id);
+        CREATE INDEX IF NOT EXISTS oauth2_refresh_tokens_user_id_idx ON oauth2_refresh_tokens(user_id);
+        CREATE INDEX IF NOT EXISTS oauth2_refresh_tokens_expires_at_idx ON oauth2_refresh_tokens(expires_at);
+
+        INSERT INTO oauth2_clients (client_id, client_secret, redirect_uris, grants, scopes, client_name) 
+        VALUES (
+            'lightdash-cli',
+            'secret',
```

Comment:
we could keep this `secret` empty, but it seems some oauth methdos like "refresh_token" enforce a secret, otherwise, client can't be validated. So I added this one here, still, most actions don't need a secret.

Reply:
If you change the getClient model to inject fixed clients, you don't need to seed the table here with a value, which is good. I don't see a good dev experience for securely inserting a secret into the database here

Reply:
all other tables for oauth reference a client to be in db. Returning afixed client on the service will not work with all these foreign keys (I tried). It is possible to remove the client reference, but that will make adding more clients harder/not compatible with oauth.